Example/tvOSSample/tvOSSample/DatabaseViewController.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import FirebaseDatabase
+
+/// A class to demonstrate the power of the Firebase Realtime Database. This will show a number read
+/// from the Database and increase or decrease it based on the buttons pressed.
+class DatabaseViewController: UIViewController {
+  private enum Counter: Int {
+    case increment = 1
+    case decrement = -1
+  }
+
+  // MARK: - Interface
+
+  /// Label to display the current value.
+  @IBOutlet weak var currentValue: UILabel!
+
+  // MARK: - User Actions
+
+  /// The increment button was hit.
+  @IBAction func incrementButtonHit(_ sender: UIButton) { changeServerValue(with: .increment) }
+
+  /// the decrement button was hit.
+  @IBAction func decrementButton(_ sender: UIButton) { changeServerValue(with: .decrement) }
+
+  // MARK: - Internal Helpers
+
+  /// Update the number on the server by a particular value. Note: the number passed in should only
+  /// be
+  private func changeServerValue(with type: Counter) {
+    let ref = Database.database().reference(withPath: Constants.databasePath)
+    // Update the current value of the number!
+    ref.runTransactionBlock { (currentData) -> TransactionResult in
+      guard let value = currentData.value as? Int else {
+        return TransactionResult.abort()
+      }
+
+      currentData.value = value + type.rawValue
```

Comment:
I don't write much Swift, but it seems a little strange to me to use the index value of an enum in a computation.

Reply:
I've used enums a bit like this since they're a lot more flexible and not limited to just a raw value anymore. It could be expanded in the future to have different values, it just saves the `switch` statement. 

I added an `intValue` calculated property that returns the `rawValue` internally, so it's more of an implementation detail as opposed to relying on the functionality externally. Does that make more sense?

Reply:
Yes, it does. Thanks for cleaning this up.